### PR TITLE
Feature/2049 flagging applications

### PR DESCRIPTION
**Pull request by LAKSHYA0501**
Targets #2049 

### Description
1. created the flag which will open a modal to get resume response data.
2. made a new column in hr_applications table which turns 0=> 1 on submission of data
3. created a new view with tables of the resumes with resume-flag value and is_desired_resume =1 

### Checklist:

- [x] I have performed a self-review of my own code.

### CODE REVIEW

**Review by P4NK4J — CHANGES_REQUESTED** [review]
The desired resume page should have title saying `Desired Resume for JOB_ROLE`. Also, the success message right now, is very faded and not catching attention on the UI, can we make it like project edit message ?
image

**Comment by LAKSHYA0501** [comment]
@P4NK4J done with the changes

**Comment by LAKSHYA0501** [comment]
Changed the table title with job roles:
image

Toast message added:
image

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php [hunk 1]
```diff
@@ -162,6 +163,7 @@ public function edit($id)
         $job = $application->job;
         $approveMailTemplate = Setting::getApplicationApprovedEmail();
         $offerLetterTemplate = Setting::getOfferLetterTemplate();
+        $resumeData = DB::table('hr_applications')->select(['hr_applications.resume'])->where('hr_applications.hr_job_id', '=', $job->id)->where('is_desired_resume', '=', 1)->get();
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:166]: can we rename it to desiredResume ?

File: Modules/HR/Http/Controllers/Recruitment/JobController.php [hunk 2]
```diff
@@ -143,4 +147,33 @@ public function storeJobdomain(JobDomainRequest $request)
...
+
+    public function showTable(HttpRequest $request)
+    {
+        $data = DB::table('hr_applications')
+            ->select(['hr_applications.resume', 'hr_application_meta.value', 'hr_jobs.title', 'hr_applicants.name'])
+            ->join('hr_application_meta', 'hr_applications.id', '=', 'hr_application_meta.hr_application_id')
+            ->join('hr_jobs', 'hr_applications.hr_job_id', '=', 'hr_jobs.id')
+            ->join('hr_applicants', 'hr_applicants.id', '=', 'hr_applications.hr_applicant_id')
+            ->where('hr_applications.hr_job_id', '=', $request->id)
+            ->where('hr_application_meta.key', '=', 'resume-flag')
```
Comment (P4NK4J) [inline Modules/HR/Http/Controllers/Recruitment/JobController.php:172]: can we rename the meta key to `reasons_for_desired_resume` ?

File: resources/views/hr/application/edit.blade.php [hunk 3]
```diff
@@ -1037,6 +1038,18 @@ class="fa fa-thumbs-down text-theme-gray-light hover-text-danger checked-text-da
...
                                                                     </div>
                                                                 </div>
+                                                                @if (!$resumeData->isempty())
+                                                                <div class="form-group col-md-5">
+                                                                    <label
+                                                                        class="text-secondary fz-14 leading-none mb-0.16">Resume
+                                                                        Link</label>
+                                                                    <div>
+                                                                        <a href="{{ route('desired.resume', [str_slug($application->job->title), $application->job->id]) }}"
+                                                                            target="_blank">sample desired resume
```
Comment (P4NK4J) [inline resources/views/hr/application/edit.blade.php:1048]:
```suggestion
                                                                            target="_blank">Sample Desired Resume
```

File: resources/views/hr/application/edit.blade.php [hunk 4]
```diff
@@ -1306,4 +1321,29 @@ class="form-control w-42p"
...
+                    </div>
+                    <div class="modal-footer">
+                        <button type="button" class="btn btn-secondary" data-dismiss="modal">Close</button>
+                        <button type="submit" class="btn btn-primary" id="responseBtn">Submit</button>
+                    </div>
+                </form>
+            </div>
+        </div>
+    </div>
+
```
Comment (P4NK4J) [inline resources/views/hr/application/edit.blade.php:1348]: Let's put this in a separate modal file and include it here

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, else looks good.

File: Modules/HR/Http/Controllers/Recruitment/ApplicationController.php (hunk 1 above)
Comment (LAKSHYA0501) [reply Modules/HR/Http/Controllers/Recruitment/ApplicationController.php:166]: Done

File: Modules/HR/Http/Controllers/Recruitment/JobController.php (hunk 2 above)
Comment (LAKSHYA0501) [reply Modules/HR/Http/Controllers/Recruitment/JobController.php:172]: doen

File: resources/views/hr/application/edit.blade.php (hunk 3 above)
Comment (LAKSHYA0501) [reply resources/views/hr/application/edit.blade.php:1048]: done

File: resources/views/hr/application/edit.blade.php (hunk 4 above)
Comment (LAKSHYA0501) [reply resources/views/hr/application/edit.blade.php:1348]: done

File: resources/views/hr/application/resume-table.blade.php [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            <thead class="thead-dark">
+                <th><strong>Resume</strong></th>
+                <th><strong>Reasons for desirability</strong></th>
+            </thead>
+            @foreach ($data as $datas)
+                <tr>
+                    <td><a href="{{ $datas->resume }}" target="_blank"><i class="fa fa-file"> {{ $datas->name }} </i></a></td>
+                    <td>{{ $datas->value }}</td>
+                </tr>
+            @endforeach
```
Comment (P4NK4J) [inline resources/views/hr/application/resume-table.blade.php:20]: What is datas, can you pass applicationData from controller and each single item can be named as $data here ?

File: resources/views/hr/application/desired-resume.blade.php [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h3 class="modal-title">Give Reason:</h3>
+                <button type="button" class="close" data-dismiss="modal">&times;</button>
+            </div>
+            <form action="{{ route('response.store', $application->id) }}" method="POST" id="responseForm">
+                @csrf
+                <div class="modal-body">
+                    <div class="fomodal-body">
```
Comment (P4NK4J) [inline resources/views/hr/application/desired-resume.blade.php:11]: What is fomodal-body ? can this classname be more specific ?

**Review by P4NK4J — CHANGES_REQUESTED** [review]
Left some comments, please check

File: resources/views/hr/application/resume-table.blade.php (hunk 5 above)
Comment (LAKSHYA0501) [reply resources/views/hr/application/resume-table.blade.php:20]: ok

**Review by P4NK4J — APPROVED** [review]
Looks good.

**Pull request merged**